src/odemis/driver/xt_client.py

```diff
@@ -349,20 +468,579 @@ def set_autofocusing(self, channel, state):
+        raise ValueError: if the argument is incorrect
+        """
+        for axis, val in shift.items():
+            if axis in self.axes:
+                axis_def = self.axes[axis]
+                if (hasattr(axis_def, "range") and
+                        abs(val) > abs(axis_def.range[1] - axis_def.range[0])):
+                    # we cannot check more precisely, unless we also know all
+                    # the moves queued (eg, if we had a targetPosition)
+                    rng = axis_def.range
+                    raise ValueError("Move %s for axis %s outside of range %f->%f"
+                                     % (val, axis, rng[0], rng[1]))
+                elif hasattr(axis_def, "choices"):
+                    # TODO: actually, in _some_ cases it could be acceptable
+                    # such as an almost continuous axis, but with only some
+                    # positions possible
+                    logging.warning("Change of enumerated axes via .moveRel() "
+                                    "are discouraged (axis %s)" % (axis,))
+            else:
+                raise ValueError("Unknown axis %s" % (axis,))
+
+    def _doMoveAbs(self, future, pos):
+        self._moveTo(future, pos)
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        Move the stage the given position in meters. This is non-blocking.
+        Throws an error when the requested position is out of range.
+
+        Parameters
+        ----------
+        pos: dict(string->float)
+            Absolute position to move the stage to per axes in m. Axes are 'x' and 'y'.
+        """
+        if not pos:
+            return model.InstantaneousFuture()
+        self._checkMoveAbs(pos)
+        pos = self._applyInversion(pos)
+
+        f = self._createFuture()
+        f = self._executor.submitf(f, self._doMoveAbs, f, pos)
+        return f
+
+    def _checkMoveAbs(self, pos):
+        """
+        Check that the argument passed to moveAbs() is (potentially) correct
+        pos (dict string -> float): the new position for a moveAbs()
+        raise ValueError: if the argument is incorrect
+        """
+        for axis, val in pos.items():
+            if axis in self.axes:
+                axis_def = self.axes[axis]
+                if hasattr(axis_def, "choices") and val not in axis_def.choices:
+                    raise ValueError("Unsupported position %s for axis %s"
+                                     % (val, axis))
+                elif (hasattr(axis_def, "range") and not
+                      axis_def.range[0] <= val <= axis_def.range[1]):
+                    # TODO: if not referenced, double the range
+                    rng = axis_def.range
+                    raise ValueError("Position %s for axis %s outside of range %f->%f"
+                                     % (val, axis, rng[0], rng[1]))
+            else:
+                raise ValueError("Unknown axis %s" % (axis,))
```

Comment:
Any reason that you overriding `_checkMoveAbs()` and `_checkMoveRel()`? They are part of `Actuator`, so they should be already available (and they look very similar).

Reply:
No this was a mistake, I'll remove them.